ipatests/test_integration/test_otp.py

```diff
@@ -398,6 +460,113 @@ def test_2fa_disable_single_prompt(self):
             self.master.run_command(['semanage', 'login', '-D'])
             sssd_conf_backup.restore()
 
+    def test_2fa_only_with_password(self):
+        """Test ssh with 2FA only with the password(first factor) when
+        user-auth-type is opt and password.
+
+        Test for :  https://github.com/SSSD/sssd/pull/7500
+
+        Add the IPA user and user-auth-type set to opt and password.
+        Authenticate the user only with password, just press enter
+        at `Second factor`
+        """
+        master = self.master
+        USER3 = 'sshuser3'
+        sssd_conf_backup = tasks.FileBackup(master, paths.SSSD_CONF)
+        first_prompt = 'Enter first factor:'
+        second_prompt = 'Enter second factor:'
+        add_contents = textwrap.dedent('''
+            [prompting/2fa/sshd]
+            single_prompt = False
+            first_prompt = {0}
+            second_prompt = {1}
+            ''').format(first_prompt, second_prompt)
+        set_sssd_conf(master, add_contents)
+        tasks.create_active_user(master, USER3, PASSWORD)
+        tasks.kinit_admin(master)
+        master.run_command(['ipa', 'user-mod', USER3, '--user-auth-type=otp',
```

Comment:
Even though you don't need to generate an OTP token with totp.generate, I think you have to create an OTP token for the user with add_otptoken(). Otherwise the ssh prompt contains only "Password:" and not the 2 prompts.